lib/CRMS/RightsPredictor.pm

```diff
@@ -0,0 +1,140 @@
+package CRMS::RightsPredictor;
+
+# Predict copyright term for Commonwealth countries based on
+# author death date or publication date.
+# Used in Commonwealth project UI for populating and updating UI with
+# rights based on metadata and researcher-provided data.
+# Also used for New Year rights rollover.
+
+# This could be extended to non-Commonwealth countries.
+
+# TODO: this code relies heavily on publication/copyright date,
+# and as a result makes Bib API calls multiple times in the course
+# of a single review. Extending the Metadata module to use a local
+# database cache for volumes in queue would enhance responsiveness
+# an eliminate some unnecessary "paranoia" code.
+
+use strict;
+use warnings;
+use utf8;
+
+use Carp;
+use POSIX;
+
+sub new {
+  my ($class, %args) = @_;
+  my $self = bless {}, $class;
+  $self->{$_} = $args{$_} for keys %args;
+  Carp::confess "No Metadata object passed to RightsPredictor" unless $args{'record'};
+  return $self;
+}
+
+# Last year the work was/will be in copyright in country of publication.
+# Returns the following structure:
+#   my $ldc = {
+#     'year' => 1933, # 4-digit int or undef on error
+#     'desc' => 'how derived'
+#   };
+# or undef on error or not enough info.
+sub last_domestic_copyright
```

Comment:
"domestic" is perhaps confusing - from whose point of view is "domestic"? If this is a term of art in copyright that stakeholders / reviewers understand to refer to the original country of publication, then it's probably fine to keep. Otherwise, perhaps consider a different name?

Reply:
Going to try "source" which may be a term of art and seems to translate into code better than "non-US."